pkg/controller/backup/backup_controller.go

```diff
@@ -263,3 +256,348 @@ func (c *Controller) enqueueBackup(obj interface{}) {
+			}
+		}
+	}
+	return false, "", nil
+}
+
+// retrySnapshotBackupAccordingToBackoffPolicy retry snapshot backup according to spec.backoffRetryPolicy.
+// the main logic is reentrant:
+//  1. check whether is retrying which is marked as BackupRetryFailed,
+//     if true, clean job and mark RealRetryAt which means current retry is done.
+//  2. check whethe is retry done, skip.
+//  3. check whether exceed retry limit, if true, mark as failed.
+//  4. check whether exceed the retry interval which is recorded as ExpectedRetryAt,
+//     the value is the time detect failure + MinRetryDuration << (retry num -1),
+//     if true, mark as BackupRetryFailed, if not, wait to next loop.
+//  5. after mark as BackupRetryFailed, the logic will back to 1 in next loop.
+func (c *Controller) retrySnapshotBackupAccordingToBackoffPolicy(backup *v1alpha1.Backup) error {
+	if len(backup.Status.BackoffRetryStatus) == 0 {
+		return nil
+	}
+	var (
+		ns           = backup.GetNamespace()
+		name         = backup.GetName()
+		now          = time.Now()
+		err          error
+		failedReason string
+		retryRecord  = backup.Status.BackoffRetryStatus[len(backup.Status.BackoffRetryStatus)-1]
+	)
+	klog.V(4).Infof("retry backup %s/%s, retry reason %s", ns, name, retryRecord.RetryReason)
+
+	// retrying
+	if isBackoffRetrying(backup) {
+		return c.doRetryFailedBackup(backup)
+	}
+
+	// retry done
+	if isCurrentBackoffRetryDone(backup) {
+		return nil
+	}
+
+	// check is exceed retry limit
+	isExceedRetryTimes := isExceedRetryTimes(backup)
+	if isExceedRetryTimes {
+		failedReason = fmt.Sprintf("exceed retry times, max is %d, original failed reason %s", backup.Spec.BackoffRetryPolicy.MaxRetryTimes, retryRecord.RetryReason)
+	}
+	isRetryTimeout := isRetryTimeout(backup, &now)
+	if isRetryTimeout {
+		failedReason = fmt.Sprintf("retry timeout, max is %d minutes, original failed reason %s", backup.Spec.BackoffRetryPolicy.RetryTimeout, retryRecord.RetryReason)
+	}
+
+	if isExceedRetryTimes || isRetryTimeout {
+		klog.Errorf("backup %s/%s exceed retry limit, failed reason %s", ns, name, failedReason)
+		err = c.control.UpdateStatus(backup, &v1alpha1.BackupCondition{
+			Type:    v1alpha1.BackupFailed,
+			Status:  corev1.ConditionTrue,
+			Reason:  "AlreadyFailed",
+			Message: failedReason,
+		}, nil)
+		if err != nil {
+			klog.Errorf("Fail to update the condition of backup %s/%s, %v", ns, name, err)
+			return err
+		}
+		return nil
+	}
+
+	// check is time to retry
+	if !isTimeToRetry(backup, &now) {
+		klog.V(4).Infof("backup %s/%s is not the time to retry, expected retry time is %s, now is %s", ns, name, retryRecord.ExpectedRetryAt, now)
+		return nil
+	}
+
+	klog.V(4).Infof("backup %s/%s is the time to retry, expected retry time is %s, now is %s", ns, name, retryRecord.ExpectedRetryAt, now)
+
+	// start to retry
+	err = c.control.UpdateStatus(backup, &v1alpha1.BackupCondition{
+		Type:    v1alpha1.BackupRetryFailed,
```

Comment:
v1alpha1.BackupRetryFailed => v1alpha1.RetryFailedBackup

Reply:
`v1alpha1.BackupRetryFailed` is the same style as [BackupConditionType](https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L1997-L2019)，it may be ok too.